### PR TITLE
util/interval: update doc.go for bazel

### PR DESCRIPTION
**Pull request by ajwerner**
Relates to #118490.

Epic: None

Release note: None

### CODE REVIEW

**Comment by ajwerner** [comment]
@nvanbenschoten maybe you're interested in this one.

File: pkg/util/interval/generic/doc.go [hunk 1]
```diff
@@ -24,7 +24,12 @@ Users of the package should follow these steps:
     internal/contract.go.
  3. Include a go generate declaration that invokes the gen.sh script with the
     type name as the first argument and the package name as the second argument.
- 4. Invoke go generate.
+    This is done for documentation purposes and will require a corresponding entry
```
Comment (ajwerner) [inline pkg/util/interval/generic/doc.go:27]: TBH I'm not 100% certain why we do this; I'm just documenting the procedure as I understand it.

**Comment by ajwerner** [comment]
@rickystewart maybe you'll like this one too? Feel free to bors it if you like it.

File: pkg/util/interval/generic/doc.go (hunk 1 above)
Comment (rickystewart) [reply pkg/util/interval/generic/doc.go:27]: The `go:generate` stanzas are generally "legacy" bits of code. Generally the `go:generate` pre-dated the existence of the Bazel code generation. There isn't an expectation that one be added assuming you are writing new code.

**Review by rickystewart — CHANGES_REQUESTED** [review]

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118491)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rickystewart)

___
*[`pkg/util/interval/generic/doc.go` line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118491#gh-1471788278:-NpVQUzC7DI0-v3PzpYF:b-v6s1o3) ([raw file](https://github.com/cockroachdb/cockroach/blob/324dc3ac5ea6d4622e166c2176cc468abf24296a/pkg/util/interval/generic/doc.go#L27)):*
<details><summary><i>Previously, rickystewart (Ricky Stewart) wrote…</i></summary><blockquote>

The `go:generate` stanzas are generally "legacy" bits of code. Generally the `go:generate` pre-dated the existence of the Bazel code generation. There isn't an expectation that one be added assuming you are writing new code.
</blockquote></details>

Removed.

**Comment by ajwerner** [comment]
@rickystewart want to merge this now?

**Review by rickystewart — APPROVED** [review]

**Comment by ajwerner** [comment]
@rickystewart I no longer have the power to bors, so if you'd give 'er a `bors r+` it'd be appreciated.

**Pull request merged**